cynic-codegen/src/fragment_derive/mod.rs

```diff
@@ -385,12 +387,17 @@ impl FragmentImpl {
                         recurse_limit: field.recurse.as_ref().map(|limit| **limit),
                     })
                 } else {
+                    let candidates: Vec<String> =
+                        object.fields.keys().map(|k| k.graphql_name()).collect();
+                    let guss_value =
+                        guess_field(candidates.borrow(), field_name.graphql_name().borrow(), 3);
```

Comment:
Once you've changed `guess_field` to take an `Iterator` you'll need to change this a bit.  (You can also usually just use the `&` operator instead of calling `.borrow()`, which I've done here)

```suggestion
                    let candidates =
                        object.fields.keys().map(|k| k.graphql_name());
                    let guss_value =
                        guess_field(candidates, field_name.graphql_name(), 3);
```

You may need to change `graphql_name` to return a `&str` instead of a `String` for this to work - I think that's a good change to make (let me know if you need a hand updating the places it's currently being used - you probably just need to add a `.clone()` or `.to_string()` on the end of them).

Reply:
Cannot expose a reference to a temporary variable,
```

391 |                     let candidates= object.fields.keys().map(|k| k.graphql_name().clone().as_str());
    |                                                                  ------------------------^^^^^^^^^
    |                                                                  |
    |                                                                  returns a value referencing data owned by the current function
    |                                                                  temporary value created here

```

Reply:
If you pull in master I've updated `graphql_name` to return a `&str` so you should be able to use (roughly) my original suggestion.